src/models/variableType.ts

```diff
@@ -1,6 +1,7 @@
 "use strict";
 
 export enum VariableType {
+    AadToken,
```

Comment:
I couldn't find any usage of VariableType other than assigning. I thought that this might be used to distinguish between VariableProcessors somewhere.

Reply:
I mean each variable provider represents one different type of variable. System variable means built-in variables, file variables mean variables defined in current file, environment variables mean variables defined in setting file and shared across all the `.http` files, and request variables are used to represent a request in a file. So a new variable type named `AadToken` seems belonging to the **System Variable** category

Reply:
> I mean each variable provider represents one different type of variable. System variable means built-in variables, file variables mean variables defined in current file, environment variables mean variables defined in setting file and shared across all the `.http` files, and request variables are used to represent a request in a file. So a new variable type named `AadToken` seems belonging to the **System Variable** category

@rafaltra any update on this comment

Reply:
That's true. I couldn't find good reason to keep it under System Variable, but it is up to you how you want to keep it :)

Reply:
Thanks @rafaltra for your quick response, I want to know why you move aad variable to a separate file, not leaving in the original system variable file

Reply:
The main thought for moving aad outside is complexity of the code inside system variable class. Most of methods were only for aad and it was looking obvious that moving it outside would be best. I have some changes in mind to improve security of aad tokens and performance of the solution. I have some bug fixes as well, but I was to close that support for username and password first.

Reply:
@rafaltra I agree with most of your opinion, however, I think we can move the aad logic to a separate file and keep the aad token variable registration in the system variable file.